codersdk/client.go

```diff
@@ -103,12 +103,12 @@ type Client struct {
 	// default 'Coder-Session-Token' is used.
 	SessionTokenHeader string
 
-	// Logger is optionally provided to log requests.
+	// logger is optionally provided to log requests.
 	// Method, URL, and response code will be logged by default.
-	Logger slog.Logger
+	logger slog.Logger
 
-	// LogBodies can be enabled to print request and response bodies to the logger.
-	LogBodies bool
+	// logBodies can be enabled to print request and response bodies to the logger.
+	logBodies bool
```

Comment:
> This treatment could theoretically be extended to other fields, but some fields (like URL) are very frequently accessed, so such a change has a large blast radius.

I think the URL can do without a mutex, it's set in `New` and shouldn't change after the fact. We probably should pass logger in `New` as well since it's a bit weird that it changes later.

PS. I wouldn't be too worried about a mutex and frequent field access, the client isn't performance critical and the delays seem negligible to me.